scripts/ci/test-coderabbit-review-gate.py

```
@@ -207,6 +207,55 @@ def test_content_and_metadata_are_identity_inputs(self):
+
+            run(["git", "-C", str(repo), "checkout", "-qb", "feature"], check=True)
+            (repo / "one.txt").write_text("one\n")
+            run(["git", "-C", str(repo), "add", "."], check=True)
+            run(["git", "-C", str(repo), "commit", "-qm", "first feature change"], check=True)
+            incremental_base = run(["git", "-C", str(repo), "rev-parse", "HEAD"], check=True, text=True, capture_output=True).stdout.strip()
+            (repo / "two.txt").write_text("two\n")
+            run(["git", "-C", str(repo), "add", "."], check=True)
+            run(["git", "-C", str(repo), "commit", "-qm", "reviewed feature head"], check=True)
+            reviewed_head = run(["git", "-C", str(repo), "rev-parse", "HEAD"], check=True, text=True, capture_output=True).stdout.strip()
+
+            run(["git", "-C", str(repo), "checkout", "-q", "master"], check=True)
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)